the-collab-lab/tcl-11-smart-shopping-list
# Feature(Issue #12) Sort shopping list view

## Description

- Edited `<Listener />` component to sort items based on how soon they're expected to be purchased. If multiple items are expected to be purchased on the same day, they'll be sorted alphabetically.
- Edited `<Item />` component to assign a different background color depending on what of the following states, the item is: Soon, Kind of Soon, Not Soon, Inactive.
- Used an `aria-label` on `<Item />` to describe each element based on `nextPurchaseInterval`.

## Related Issue

Closes #12 

## Acceptance Criteria


* Items in the list are shown as visually distinct (e.g., with a different background color on the list item) according to how soon the item is expected to be bought again: Soon, Kind of soon, Not soon, Inactive
* Items should be sorted by the estimated number of days until next purchase
* Items with the same number of estimated days until next purchase should be sorted alphabetically
* Items in the different states should be described distinctly when read by a screen reader

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
| ✓   | :sparkles: New feature     |
|  ✓   | :hammer: Refactoring       |


## Updates

### Before

#### Unsorted Items
![image](https://user-images.githubusercontent.com/6282435/92793064-16121700-f37c-11ea-9935-4564a41b0daa.png)


### After



#### Sorted + Color Coded Items
![image](https://user-images.githubusercontent.com/6282435/92792673-c0d60580-f37b-11ea-8f5e-6821d8ae85c6.png)


## Testing Steps / QA Criteria

- From your terminal, run git checkout `bl-vh-shopping-list-view`
- Run `npm install` to ensure that your dependencies are up to date
- Run `npm run start`
- Visit the component at `http://localhost:3000`
- Visit a shopping list, making note of the list's token
- Visit [https://console.firebase.google.com/u/0/project/tcl-11-smart-shopping-list](https://console.firebase.google.com/u/0/project/tcl-11-smart-shopping-list) and go to `cloud firestore`
- When in firestore, visit your list's collection, and edit the items' `nextPurchaseInterval`, and `numberOfPurchases` (`numberOfPurchases` is important here, because by default, these items would be detected as inactive).
- Return to `http://localhost:3000`
- Your items will now be sorted in ascending order based on `nextPurchaseInterval` 💃
- If multiple items have the same `nextPurchaseInterval`, they will be sorted alphabetically.
- In addition to being sorted, your items will be color-coded depending on item state (Soon, etc).
